test/e2e/app/app.go

```diff
@@ -204,6 +204,11 @@ func (app *Application) ApplySnapshotChunk(req abci.RequestApplySnapshotChunk) a
 	return abci.ResponseApplySnapshotChunk{Result: abci.ResponseApplySnapshotChunk_ACCEPT}
 }
 
+func (app *Application) PrepareProposal(
+	req abci.RequestPrepareProposal) abci.ResponsePrepareProposal {
+	return abci.ResponsePrepareProposal{BlockData: req.BlockData}
+}
```

Comment:
Orthogonal to this PR and I'm not sure if this should be done in the tendermint or on the SDK side but eventually, it would be good if the added abci methods would also be tested in a more meaningful way in e2e tests (with a method that actually changes Block.Data).

Reply:
good point, I can include this in the pr